test/lib/templateFactory.test.js

```diff
@@ -122,4 +123,69 @@ describe('Template Factory', () => {
                 Error, 'No datastore provided to TemplateFactory');
         });
     });
+
+    describe('getTemplate', () => {
+        let config;
+        let expected;
+        let returnValue;
+
+        beforeEach(() => {
+            config = {
+                name: 'testTemplateName',
+                version: '1.0'
+            };
+
+            returnValue = [
+                {
+                    id: '151c9b11e4a9a27e9e374daca6e59df37d8cf00f',
+                    name: 'testTemplateName',
+                    version: '1.0.1',
+                    labels: ['testLabel', 'otherLabel']
+                },
+                {
+                    id: 'd398fb192747c9a0124e9e5b4e6e8e841cf8c71c',
+                    name: 'testTemplateName',
+                    version: '1.0.3',
+                    labels: ['otherLabel']
+                },
+                {
+                    id: '151c9b11e4a9a27e9e374daca6e59df37d8cf00f',
+                    name: 'testTemplateName',
+                    version: '1.0.2',
+                    labels: ['testLabel', 'otherLabel']
+                },
+                {
+                    id: '151c9b11e4a9a27e9e374daca6e59df37d8cf00f',
+                    name: 'testTemplateName',
+                    version: '2.0.1',
+                    labels: ['testLabel', 'otherLabel']
+                }
+            ];
+        });
+
+        it('should get the correct template for a given config with label', () => {
```

Comment:
It will return undefined. According to the doc for find: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find